pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -282,6 +282,15 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("A volume match does not exist for persistent claim: %s", claim.Name)
 		}
 
+		// create a reference to the claim and assign it to the volume being bound.
+		// the volume is a pointer and assigning the reference fixes a race condition where another
+		// claim might match this volume but before the claimRef is persistent in the next case statement
+		claimRef, err := api.GetReference(claim)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+		}
+		volume.Spec.ClaimRef = claimRef
```

Comment:
@derekwaynecarr i can remove the race entirely by changing the binding behavior.  By putting it on pv.Spec.ClaimRef, there is no race and subsequent `syncClaim()` calls are idempotent.  This changes undocumented behavior, though. Is there a precedent for that?